### ISSUE TITLE
Sync with other maps apps (Favorites, Tracks, etc.)

### ISSUE THREAD

**Issue by jancborchardt** [nextcloud/maps#92]
We should work on having these apps sync with us:
- [ ] [GNOME Maps](https://wiki.gnome.org/Apps/Maps) might become compatible this summer, ref **Provide APIs to sync with gnome-maps #30**, cc @Sumaid @mlundblad
- [ ] [Marble](https://marble.kde.org/) would require a few changes to be compatible
- [ ] [OsmAnd](https://osmand.net/) would be nice! Nothing like that yet.
- [ ] [Maps.me](https://maps.me/) would be best since they are widely used, but they have a company behind them and already offer their own proprietary favorites backup.

(Carried over from @eneiluj’s notes in https://github.com/nextcloud/maps/issues/56 )

**Comment by thomas-mc-work** [comment]
There is a very actively developed clone of Maps.me called [Maps](https://gitlab.com/axet/omim). Maybe this would be a good candidate for integration? They have already implemented track recording which would play nicely with the sync feature.

> Features over upstream repo:
> 
> - Track Recording (record your tracks)
> - Custom Bookmarks path (helps backup your bookmarks with apps like: Syncthing)
> - Removed ads and binaries (peace and freedom)
> - Screenlock option (prevent app stay turned on while in pocket with lockscreen enabled)

**Comment by jancborchardt** [comment]
@thomas-mc-work good call, yes! Do you have any contacts there or do you want to open an issue in their tracker linking both projects up? :)

**Comment by thomas-mc-work** [comment]
I've raised an issue [here](https://gitlab.com/axet/omim/issues/122), mostly copied from the OsmAnd issue and hopefully phrased as expected.

Where can I find the API documentation? Preferably in the [OpenAPI format](https://github.com/OAI/OpenAPI-Specification/blob/master/versions/3.0.2.md).

**Comment by eneiluj** [comment]
@thomas-mc-work Actually there is no Maps Favorites API documentation. It's pretty straightforward though. It's a REST API with 4 routes (listed in `appinfo/routes.php`):

* GET https://mync.org/index.php/apps/maps/api/1.0/favorites => get a JSON list of the favorites
* POST https://mync.org/index.php/apps/maps/api/1.0/favorites => add a favorite, parameters are name (string), lat (float), lng (float), category (string), comment (string), extensions (string)
* PUT https://mync.org/index.php/apps/maps/api/1.0/favorites/ID => with a value instead of "ID", edit a favorite. Params are the same as POST ones.
* DELETE https://mync.org/index.php/apps/maps/api/1.0/favorites/ID => with a value instead of "ID", deletes a favorite

For accessing those routes, you need to be authenticated with a session or with basic http auth.

I'm completely open to make changes to this API since it's not used yet :wink:.

**Comment by rinigus** [comment]
I wonder whether favorites fields are fixed now. There are some extra fields that I would like to have for Pure Maps or are those expected to be stored under "extensions"?

As I don't have any experience with Nextcloud sync - what would be the best way to authenticate for access? I would prefer not to store any passwords - some kind of session id? Maybe pointers towards docs on it?

**Comment by szaimen** [comment]
Maybe this?
https://github.com/nextcloud/Android-SingleSignOn

**Comment by rinigus** [comment]
@szaimen : thanks, but I am looking for something running on Linux (desktop and mobile), not Android.

**Comment by gdepountis** [comment]
I am sorry if I am not in correct conversaion but
does maps are compatible with the marinetraffic API
https://www.marinetraffic.com/en/ais-api-services/detail/ps02/vessel-positions-of-a-static-fleet#

**Comment by jancborchardt** [comment]
@gdepountis could you open a separate issue for that? :) Thanks

**Comment by bugzy** [comment]
What changes would marble require to become compatible?

**Comment by tacruc** [comment]
@bugzy I had a look at the marble code, unfortunately I have only very basic c and c++ skills and none in QT. For me it looks like the was hardcoded in some way (.../apps/MARBLEAPP/...), so for sure the URL's needs to be adapted.  But I guess than a lot of code could be reused.

**Comment by Spartachetto** [comment]
Just for information: the current release of OsmAnd introduced the option to export and import all data. For reference have a look here: https://osmand.net/blog/osmand-3-9-released .
It states that _We added the option to export and import all data including settings, resources, my places. It's a big step to the backup feature, which we will plan to add in the next release_ .

**Comment by keunes** [comment]
> Maps.me would be best since they are widely used, but they have a company behind them and already offer their own proprietary favorites backup.

@jancborchardt, all: There now is a fork of Maps.me in development: [Organic Maps](https://github.com/organicmaps/organicmaps) - I reckon it could be added to the list in the first post (or replace the Maps.me list item). I would love to have synchronisation between the two, and added [synchronisation request over at Organic Maps](https://github.com/organicmaps/organicmaps/discussions/776) as well.

**Comment by penguin86** [comment]
I recently switched to Nextcloud and have the urgent need to open my imported geobookmarks in Organic Maps. I tried to clone the OM sources and build the app, just to have an idea of the complexity of the implementation, but didn't succeed... I'm a little rusty at Android development.
So **I developed a very basic app that allows to open the Nextcloud Maps geobookmarks in any android app** supporting Geo URI (e.g. Organic Maps, Google Maps...). Don't expect bells and whistles, it's the bare minimum to get me on the road again, but I hope it can be useful to someone.

You can download from [the github page](https://github.com/penguin86/nextcloud-maps-client/releases) and soon will be available on [F-Droid](https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9655).

I home the source can be useful as an example to integrate Nextcloud Geobookmarks support in any of the proposed apps.

**Comment by szaimen** [comment]
Very cool! Do you mind opening a PR in the maps repository that adds it to the readme? Would definitely improve the discoverability :)

**Comment by penguin86** [comment]
> Very cool! Do you mind opening a PR in the maps repository that adds it to the readme? Would definitely improve the discoverability :)

Thank you @szaimen ! It's a very good idea, but I'd wait until the app is mature enough... As a user, finding the link to the app, I'd expect it to implement all the functionalities provided by Nextcloud Maps (Photos, Geobookmarks, Tracks, Contacts), at least at a basic level...

**Comment by szaimen** [comment]
Though having at least a geobookmark sync is already more than any other app is capable of, currently. So I still vote for adding it to the readme :)

**Comment by xplosionmind** [comment]
Hi! I noticed that this issue has been stale for almost two years now, even though it is the one with the most thumbs up… are there any news? It would be great to have some bookmarks integration!

**Comment by tacruc** [comment]
Well the bookmarks app is out there. https://gitlab.com/fdroid/fdroiddata/-/merge_requests/9655

**Comment by FailurePoint** [comment]
Hey, so did anyone ever develop a app or add sync to am existing one? I would love to be able to dich google maps and have it sync to my server.

**Comment by KrausMatthias** [comment]
I would suggest using https://ogcapi.ogc.org/features/ as exchange api, then you don't need to implement a sync API for every single Map App / Server

**Comment by SyndicatedPillbug** [comment]
Just really wishing for this! I use OSMAnd and it'd be So nice to automatically sync favorites and locations and photos and everything else from OSMand to nextcloud.

**Comment by g6094199** [comment]
status ping?

seems nothing really changed in the right direction in the last 6 years?
